### PR TITLE
custam proxy

### PR DESCRIPTION
**Pull request by imhamza333**

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -61,9 +60,14 @@
...
+# DOWNLOADER_MIDDLEWARES = {
+#         'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 110,
+#         'scraping.middlewares.CustomProxyMiddleware': 543,
+# }
+CustomProxyMiddleware = True
 DOWNLOADER_MIDDLEWARES = {
-        'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 110,
-        'scraping.middlewares.ScrapingDownloaderMiddleware': 543,
+    'scraping.middlewares.CustomProxyMiddleware': 350,
+    'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 400,
```
Comment (AdnanAsif) [inline settings.py]: Ye middleware kaha se ai ha ?

File: spiders/test.py [hunk 2]
```diff
@@ -1,15 +1,13 @@
 import scrapy
 
 
-class ProxiesDownloaderMiddleware(scrapy.Spider):
+class Test_spider(scrapy.Spider):
```
Comment (AdnanAsif) [inline spiders/test.py]: Follow PEP-8 guidelines to name the class

**Review by AdnanAsif — CHANGES_REQUESTED** [review]

File: settings.py (hunk 1 above)
Comment (imhamza333) [reply settings.py]: ya old wala middleware phala kaha tha jo ma na commet kar deya. pir new wala creat kar deya ha

File: spiders/test.py (hunk 2 above)
Comment (imhamza333) [reply spiders/test.py]: ok

File: settings.py (hunk 1 above)
Comment (AdnanAsif) [reply settings.py]: means it was in default setting ?

Comment (imhamza333) [reply settings.py]: yes

Comment (AdnanAsif) [reply settings.py]: By default ye middleware nai hoti settings me.

**Comment by AdnanAsif** [comment]
Where are the changes you said you have done ?

File: spiders/cclonline.py [hunk 3]
```diff
@@ -9,6 +9,17 @@ class Cclonline_spider(scrapy.Spider):
         "https://www.cclonline.com/",
     ]
 
+    custom_settings = {'RETRY_HTTP_CODES': [403],
```
Comment (AdnanAsif) [inline spiders/cclonline.py:12]: What is this for ?

File: settings.py [hunk 4]
```diff
@@ -56,18 +47,11 @@
...
 # Enable or disable downloader middlewares
 # See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
-
-# DOWNLOADER_MIDDLEWARES = {
-#         'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 110,
-#         'scraping.middlewares.CustomProxyMiddleware': 543,
-# }
-CustomProxyMiddleware = True
 DOWNLOADER_MIDDLEWARES = {
+    'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 110,
```
Comment (AdnanAsif) [inline settings.py:53]: This middleware is enabled by default, we do not need to add it here.

File: settings.py [hunk 5]
```diff
@@ -7,26 +7,17 @@
...
-# USER_AGENT = get_random_agent()
-
 BOT_NAME = 'scraping'
 
 SPIDER_MODULES = ['scraping.spiders']
 NEWSPIDER_MODULE = 'scraping.spiders'
 
 # Crawl responsibly by identifying yourself (and your website) on the user-agent
-# USER_AGENT = 'scraping (+http://www.yourdomain.com)'
+# USER_AGENT = 'newone (+http://www.yourdomain.com)'
```
Comment (AdnanAsif) [inline settings.py:16]: Why changed this ? Its a comment, changing it will do nothing.

File: settings.py [hunk 6]
```diff
@@ -79,7 +63,7 @@
 # Configure item pipelines
 # See https://docs.scrapy.org/en/latest/topics/item-pipeline.html
 # ITEM_PIPELINES = {
-#    'scraping.pipelines.ScrapingPipeline': 300,
+#    'scraping.pipelines.scrapingPipeline': 300,
```
Comment (AdnanAsif) [inline settings.py:66]: Why changed this ?

File: spiders/cclonline.py [hunk 7]
```diff
@@ -20,15 +31,14 @@ class Cclonline_spider(scrapy.Spider):
     #         yield scrapy.Request(url, meta={'proxy': 'https://51.79.166.87:8080'})
 
 
-class ProxiesDownloaderMiddleware(object):
 
-    def process_request(self, request):
-        request.meta['proxy'] = 'https://51.79.166.87:8080'
+# class ProxiesDownloaderMiddleware(object):
```
Comment (AdnanAsif) [inline spiders/cclonline.py:35]: Remove this, if not needed.

**Review by AdnanAsif — CHANGES_REQUESTED** [review]

**Comment by AdnanAsif** [comment]
@imhamza333 why merged the PR ?

File: spiders/cclonline.py (hunk 3 above)
Comment (imhamza333) [reply spiders/cclonline.py:12]: ccloin wala spider pa test kar ra tha. us ka liya ya set kia tha ager 403 ho to retry kara

File: settings.py (hunk 4 above)
Comment (imhamza333) [reply settings.py:53]: ok

File: settings.py (hunk 5 above)
Comment (imhamza333) [reply settings.py:16]: new project creat kia tha ya us key setting.py file thi jis ma project name change kia tha

File: spiders/cclonline.py (hunk 7 above)
Comment (imhamza333) [reply spiders/cclonline.py:35]: ok

**Comment by imhamza333** [comment]
> @imhamza333 why merged the PR ?

ma samja ya old wali request thi is liya new creat kar na ka liya old wali merage key

**Pull request merged**